analysis/trigger_performance/01_readiness.md

```diff
@@ -0,0 +1,400 @@
+# Readiness trigger performance
+
+Reads in the readiness analysis by Joseph, does some BSRS,
+and gets the stats
+
+```python
+%load_ext jupyter_black
+```
+
+```python
+from pathlib import Path
+import os
+import warnings
+
+import numpy as np
+import pandas as pd
+```
+
+```python
+rng = np.random.default_rng(12345)
+
+trigger_dir = (
+    Path(os.environ["AA_DATA_DIR"])
+    / "public/exploration/phl/trigger_performance"
+)
+```
+
+## Reading in the readiness data
+
+```python
+trigger_filename = trigger_dir / "Readiness trigger performance.xlsx"
+df_readiness = pd.read_excel(trigger_filename)
+df_readiness
+```
+
+```python
+# Convert scenarios to strings
+for cname in ["Activation scenario", "Readiness scenario"]:
+    df_readiness[cname] = df_readiness[cname].fillna(0).astype(int).astype(str)
+# Make sure Y/N is capitalized
+for cname in ["Should have activated", "Readiness reached"]:
+    df_readiness[cname] = df_readiness[cname].str.upper()
+```
+
+```python
+# fill in confusion matrix
+df_readiness["Confusion matrix"] = "TN"
+df_readiness.loc[
+    (df_readiness["Should have activated"] == "Y")
+    & (df_readiness["Readiness reached"] == "N"),
+    "Confusion matrix",
+] = "FN"
+df_readiness.loc[
+    (df_readiness["Should have activated"] == "Y")
+    & (df_readiness["Readiness reached"] == "Y")
+    & (
+        df_readiness["Activation scenario"]
+        == df_readiness["Readiness scenario"]
+    ),
+    "Confusion matrix",
+] = "TP"
+# Dataset has no FPs so skipping those
+```
+
+```python
+# Rename confusion matrix column
+df_readiness = df_readiness[
+    ["International name", "Year", "Confusion matrix", "Should have activated"]
+].rename(
+    columns={
+        "International name": "typhoon",
+        "Year": "year",
+        "Confusion matrix": "readiness",
+    }
+)
+# Capitlize typhoon names
+df_readiness["typhoon"] = (
+    df_readiness["typhoon"].str.upper().str.strip().str.replace(" ", "-")
+)
+
+df_readiness
+```
+
+## Activation trigger
```

Comment:
`action` in OCHA terminology

Reply:
Unfortunately in the Philippines framework it's called "activation" :facepalm: